site/public/js/forum.js

```diff
@@ -2104,3 +2104,30 @@ $(() => {
 function scrollThreadListTo(element){
   $(element).get(0).scrollIntoView({behavior: "smooth", block: "center"});
 }
+
+//Only used by the posters and only on recent posts (60 minutes since posted)
+function sendAnnouncement(title, thread_post_content, id){
+  $.ajax({
+    type: 'POST',
+    url: buildCourseUrl(['forum', 'make_announcement']),
+    data: {'title': title, "thread_post_content": thread_post_content, "id": id, 'csrf_token': window.csrfToken},
+    success: function(data){
+      alterAnnouncement(id, "Are you sure you want to pin this thread to the top?", 1, window.csrfToken);
+    },
+  });
+}
+
+function checkIfAnnounced(id){
+  $('.pin-and-email-message').hide();
+  call = $.ajax({
+    type: 'GET',
+    url: buildCourseUrl(['email', 'check_announcement']),
+    data: {'thread_id': id, "csrfToken": csrfToken},
+    success: function(res) {
+      const response = JSON.parse(res);
+      if (response.status === "success" && response["data"]["exists"] == false){
+        $('.pin-and-email-message').show();
+      }
+    }
+  });
```

Comment:
These should have some error handling whether the request failed or the backend reported an issue with input data.

Reply:
> The button works and pins the post to the top. The button appeared for me when I created a new announcement though. I think one of your queries checks for emails and is conditional based off of that. Every minute 100 emails are sent all at once. This could lead to the button showing and an instructor clicking it while it's already an announcement. Also if a thread has a question mark on it's preview, after making it an announcement it still keeps that question mark. That should probably go away.

This actually brings up an issue that I somehow didn't see before (for some reason I thought the announcement title includes the thread id). 

- An issue right now and possible ways to solve them: currently the database makes an announcement with only the thread title in the announcement and no thread id, meaning it is possible that the making the announcement button is incorrectly hidden

1. Option 1: We can change the announcement title to also include the thread id

2. Option 2: Assume the instructors/TA won't make a thread with duplicate title...

Reply:
Yeah I think option 1 is definitely the better way to go compared to option 2. Option 2 could happen completely by accident. We need to make sure that when the page loads that it doesn't still show that button cause it may look like the instructor forgot to set it as an announcement. Now there is twice the emails.